### PR TITLE
sorter/leveldb(ticdc): trigger compact by deletion to speed up read

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Add db compactor to gc delete entries and reclaim disk space.

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sorter/leveldb/cleaner.go [hunk 1]
```diff
@@ -173,6 +178,15 @@ func (clean *CleanerActor) close(err error) {
 	clean.closedWg.Done()
 }
 
+// Schedule a compact task when there are too many deletion.
+func (clean *CleanerActor) maybeScheduleCompact() {
+	if clean.compact.maybeCompact(clean.id, clean.deleteCount) {
+		// Reset delete key count if schedule compaction successfully.
+		clean.deleteCount = 0
+		return
```
Comment (hi-rustin) [inline cdc/sorter/leveldb/cleaner.go:186]: Useless return here?

File: cdc/sorter/leveldb/compactor.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		default:
+			log.Panic("unexpected message", zap.Any("message", msg))
+		}
+	}
+
+	start, end := []byte{0x0}, bytes.Repeat([]byte{0xff}, 16)
+	now := time.Now()
+	if err := c.db.Compact(start, end); err != nil {
+		log.Error("db compact error", zap.Error(err))
+	}
```
Comment (amyangfei) [inline cdc/sorter/leveldb/compactor.go:87]: Is there any side effect to ignore this error?

File: pkg/config/db.go [hunk 3]
```diff
@@ -65,6 +60,16 @@ type DBConfig struct {
...
+	// CompactionL0Trigger defines number of leveldb sst file at level-0 that will
+	// trigger compaction.
+	//
+	// The default value is 160.
+	CompactionL0Trigger int `toml:"compaction-l0-trigger" json:"compaction-l0-trigger"`
+	// CompactionDeletionThreshold defines the threshold of the number of deletion that
+	// trigger compaction.
+	//
+	// The default value is 960000.
+	CompactionDeletionThreshold int `toml:"compaction-deletion-threshold" json:"compaction-deletion-threshold"`
```
Comment (amyangfei) [inline pkg/config/db.go:72]: How is this value determined

**Review by amyangfei — APPROVED** [review]

File: cdc/sorter/leveldb/compactor.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		case actormsg.TypeTick:
+		case actormsg.TypeStop:
+			c.close(nil)
+			return false
+		default:
+			log.Panic("unexpected message", zap.Any("message", msg))
+		}
+	}
+
+	start, end := []byte{0x0}, bytes.Repeat([]byte{0xff}, 16)
```
Comment (asddongmen) [inline cdc/sorter/leveldb/compactor.go:83]: Could you add a comment here?

Comment (asddongmen) [reply cdc/sorter/leveldb/compactor.go:83]: rest LGTM

**Review by asddongmen — APPROVED** [review]

**Pull request merged**